### PR TITLE
Advent of code

### PR DESCRIPTION
**Pull request by dmitri-sirobokov**
Blog article, sharing experience with Advent Of Code 2022, https://adventofcode.com/2022/about

### CODE REVIEW

File: data/blog/advent-of-code.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+## Experience
+
+### Time
+
+The first few days when I started, it took me around 30 minutes or quicker to find the solutions. So I could just wake up a bit earlier before going to work. As complexity increased almost every day, it took me more than an hour or two. I had to use my lunch breaks for it and time in the evenings. And then there were a couple of problems that took me more than a day to solve. But roughly, I think I spent on average 2-3 hours per puzzle. That is 50-70 hours in total! That is a lot! And it won't go unnoticed. My advice, make an agreement with your friends, partner or with yourself if you want to find some time for it.
+
+### Typing skills
+
+After typing hundreds of for-loops, I wish I could think and type as quickly as [Victor Rentea](https://www.youtube.com/watch?v=cK19rE2V9UY), and I wish I watched his videos before. That would save me so much time. Something to learn for the next time!
```
Comment (DaveBitter) [inline data/blog/advent-of-code.md]: Perhaps embed the YouTube video? I believe it is possible

File: data/blog/advent-of-code.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+               +- y2022\ # Input to your puzzles
+                    |
+                    +- day1.txt         # Your real input to the puzzle Day 1
+                    +- day1_sample.txt  # Sample puzzle input for the Day 1, for which the answere is provided.
+                    +- day2.txt
+                    +- day2_sample.txt
+
+```
+
+One more thing to mention here. Starting from Day 5 I am using parameterized unit tests. I figured out how to extend my JUnit framework using ArgumentsProvider interface, I have removed duplicate code that reads sample files for my test cases. My code became at least 2x shorter and cleaner after that:
```
Comment (DaveBitter) [inline data/blog/advent-of-code.md]: Perhaps link to some of the resources you used to learn about it?

File: data/blog/advent-of-code.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+        var result = Day08.part2(input);
+
+        assertEquals(expected, result);
+    }
+}
+```
+
+## Classes of problems
+
+## Integer overflow
```
Comment (DaveBitter) [inline data/blog/advent-of-code.md]: It looks like you heading levels are off. There are two h2's after each other here

**Review by DaveBitter — COMMENTED** [review]
Great article @dmitri-sirobokov!

File: data/blog/advent-of-code.md (hunk 1 above)
Comment (dmitri-sirobokov) [reply data/blog/advent-of-code.md]: Done

File: data/blog/advent-of-code.md (hunk 2 above)
Comment (dmitri-sirobokov) [reply data/blog/advent-of-code.md]: Done

File: data/blog/advent-of-code.md (hunk 3 above)
Comment (dmitri-sirobokov) [reply data/blog/advent-of-code.md]: Fixed.

**Comment by DaveBitter** [comment]
LGTM, @sanderdejong88 ?

**Review by DaveBitter — APPROVED** [review]

**Pull request merged**